src/NuGet.Core/NuGet.Protocol/Plugins/PluginFile.cs

```diff
@@ -43,6 +43,32 @@ public PluginFile(string filePath, Lazy<PluginFileState> state, bool requiresDot
             State = state;
         }
 
+#if IS_DESKTOP
+        /// <summary>
+        /// Instantiates a new <see cref="PluginFile" /> class.
+        /// The plug-in will be executed directly.
+        /// If it needs to run under the dotnet host, use <see cref="PluginFile(string, Lazy{PluginFileState}, bool)" /> and specify <see langword="true" /> for the <c>requiresDotnetHost</c> parameter.
+        /// </summary>
+        /// <param name="filePath">The plugin's file path.</param>
+        /// <param name="state">A lazy that evaluates the plugin file state.</param>
+#else
+        /// <summary>
+        /// Instantiates a new <see cref="PluginFile" /> class.
+        /// The plug-in will be executed under the dotnet host.
+        /// If it needs to be executed directly, use <see cref="PluginFile(string, Lazy{PluginFileState}, bool)" /> and specify <see langword="false" /> for the <c>requiresDotnetHost</c> parameter.
+        /// </summary>
+        /// <param name="filePath">The plugin's file path.</param>
+        /// <param name="state">A lazy that evaluates the plugin file state.</param>
+#endif
+        public PluginFile(string filePath, Lazy<PluginFileState> state)
```

Comment:
Didn't we have this constructor in the past? :D 

https://github.com/NuGet/NuGet.Client/blob/release-6.12.x/src/NuGet.Core/NuGet.Protocol/Plugins/PluginFile.cs had this constructor. 

https://github.com/NuGet/NuGet.Client/blob/release-6.13.x/src/NuGet.Core/NuGet.Protocol/Plugins/PluginFile.cs changed it and now 6.14 is bringing it back :D 

In a way this is a "fix" for the binary breaking change we did.

Reply:
That's hilarious, great memory.  Well I guess this is a better pattern and won't break anything?

Reply:
Yeah, it won't break anything.
We weren't too worried about it breaking things because it's usually not a type you'd invoke manually. 
The situation is just hilarious :D